best-practices/go-ci.md

```diff
@@ -0,0 +1,70 @@
+## Best practices for Continuous Integration with Go
+
+First, use [Modules](https://golang.org/ref/mod) even if you don't have any
+external dependencies. Beyond ensuring consistency, a module allows using new Go
+language features. The instructions below assume you are setting up CI for a
+single Go module, and a POSIX-compatible shell like Bash.
+
+It is strongly recommended that the following checks are used:
+
+* `go test ./...` to run all tests
+* `go test -race ./...` to catch any data races when running the tests
+* `test -z $(gofmt -l .)` to ensure all Go files are well formatted
+* `go vet ./...` to ensure there are no vet warnings
+
+The following checks are also recommended, though they might not help in all
+situations:
+
+* `go mod tidy && git diff --exit-code` to ensure `go.mod` and `go.sum` are clean
+* `go mod verify` to ensure that `go.sum` agrees with what's in the module cache
+* `go test -run=- -bench=. -benchtime=1x ./...` to catch stale benchmarks
+* `staticcheck ./...` to ensure there are no [Staticcheck](https://staticcheck.io/) warnings
+
+Finally, if your module is a library, it is recommended to support at least the
+two latest major Go versions. For example, if Go 1.16.x is the latest major
+release, all the checks above should be run on 1.16.x, and `go test ./...`
+should also run on 1.15.x. Note that upstream Go supports
+[two major releases](https://github.com/golang/go/wiki/Go-Release-Cycle#release-maintenance)
+at once as well.
+
+#### How to use this document at Protocol Labs
+
+Marten Seemann and the IPLD team have set up [a template repository](https://github.com/ipld/.github)
+to share YAML workflows between IPLD repositories. The repository itself
+uses GitHub Actions to copy specific workflows into lists of repositories. For
+example, the "Go" workflows are regularly copied into a number of IPLD Go
```

Comment:
It’s not done on a schedule, but every time the template workflow is changed.

Reply:
Done.